src/jackdaw/serdes/avro.clj

```diff
@@ -399,21 +399,24 @@
 
   (avro->clj [_ avro-record]
     (when avro-record
-      (into {}
-            (comp (map first)
-                  (map (fn [^Schema$Field field]
-                         (let [field-name (.name field)
-                               field-key (keyword (unmangle field-name))
-                               [_ field-coercion :as entry] (get field->schema+coercion field-key)
-                               value (.get ^GenericData$Record avro-record field-name)]
-                           (when-not field-coercion
-                             (throw (ex-info "Unable to deserialize field"
-                                             {:field field
-                                              :field-name field-name
-                                              :field-key field-key
-                                              :entry entry})))
-                           [field-key (avro->clj field-coercion value)]))))
-            (vals field->schema+coercion))))
+      (with-meta
+        (into {}
+              (comp (map first)
+                    (map (fn [^Schema$Field field]
+                           (let [field-name (.name field)
+                                 field-key (keyword (unmangle field-name))
+                                 [_ field-coercion :as entry] (get field->schema+coercion field-key)
+                                 value (.get ^GenericData$Record avro-record field-name)]
+                             (when-not field-coercion
+                               (throw (ex-info "Unable to deserialize field"
+                                               {:field field
+                                                :field-name field-name
+                                                :field-key field-key
+                                                :entry entry})))
+                             [field-key (avro->clj field-coercion value)]))))
+              (vals field->schema+coercion))
+        {:name (.getName schema)
+         :fullname (.getFullName schema)})))
```

Comment:
I think when hyphenating the `FullName` symbol I'd expect to see `full-name`.

Reply:
I just pushed a commit with the hyphenated name.